sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -0,0 +1,523 @@
+    assetExchangeAgreement.setRecipient(recipientECert);
+    assetExchangeAgreement.setLocker(lockerECert);
+    return assetExchangeAgreement.serializeBinary().toString();
+}
+
+// Create an asset lock structure
+function createAssetLockInfoSerialized(hashValue, expiryTimeSecs)
+{
+    const lockInfoHTLC = new assetLocksPb.AssetLockHTLC();
+    lockInfoHTLC.setHashbase64(hashValue);
+    lockInfoHTLC.setExpirytimesecs(expiryTimeSecs);
+    lockInfoHTLC.setTimespec(assetLocksPb.AssetLockHTLC.TimeSpec.EPOCH)
+    const lockInfoHTLCSerialized = lockInfoHTLC.serializeBinary();
+    const lockInfo = new assetLocksPb.AssetLock();
+    lockInfo.setLockmechanism(assetLocksPb.LockMechanism.HTLC);
+    lockInfo.setLockinfo(lockInfoHTLCSerialized);
+    return lockInfo.serializeBinary().toString();
+}
+
+// Create an asset claim structure
+function createAssetClaimInfoSerialized(hashPreimage)
+{
+    const claimInfoHTLC = new assetLocksPb.AssetClaimHTLC();
+    claimInfoHTLC.setHashpreimagebase64(Buffer.from(hashPreimage).toString('base64'));
+    const claimInfoHTLCSerialized = claimInfoHTLC.serializeBinary();
+    const claimInfo = new assetLocksPb.AssetClaim();
+    claimInfo.setLockmechanism(assetLocksPb.LockMechanism.HTLC);
+    claimInfo.setClaiminfo(claimInfoHTLCSerialized);
+    return claimInfo.serializeBinary().toString();
+}
+
+// Create a SHA-256 hash over an ASCII string
+function createSHA256HashBase64(preimage)
+{
+    return crypto.createHash('sha256').update(preimage).digest('base64');
+}
+
+
+/**
+ * First/second step of a Hashed Time Lock Contract
+ * - Lock a unique asset instance using a hash
+ **/
+const createHTLC = async (
+    contract: Contract,
+    assetType: string,
+    assetID: string,
+    recipientECert: string,
+    hashValue: string,
+    expiryTimeSecs: number,
+): Promise<{ preimage: any; result: any }> => {
+
+    if (!contract)
+    {
+        logger.error("Contract handle not supplied");
+        return { preimage: "", result: false };
+    }
+    if (!assetType)
+    {
+        logger.error("Asset type not supplied");
+        return { preimage: "", result: false };
+    }
+    if (!assetID)
+    {
+        logger.error("Asset ID not supplied");
+        return { preimage: "", result: false };
+    }
+    if (!recipientECert)
+    {
+        logger.error("Recipient ECert not supplied");
+        return { preimage: "", result: false };
+    }
+
+    let defaultExpiryTimeSecs = 0, preimage = "";
+    if (hashValue && hashValue.length > 0)
+    {
+        defaultExpiryTimeSecs = 5 * 60     // 5 mins: this is the 't' used to time out the second contract of the HTLC pair
+    }
+    else
+    {
+        defaultExpiryTimeSecs = 10 * 60     // 10 mins: this is the '2t' used to time out the first contract of the HTLC pair
+
+        // Create a random preimage of 20 bytes
+        for (let i = 0 ; i < 20 ; i++)
+        {
+            preimage += String.fromCharCode(Math.floor(Math.random() * 256))
```

Comment:
should we be using a cryptographically secure pseudo-random number generator as opposed to `Math.random()`?

Reply:
Done.